### ISSUE TITLE
Unable to get default network address while attempting to stream cameras

### ISSUE THREAD

**Issue by bc2297** [dgreif/ring#544]
# Live Streaming Issue

### Describe the live streaming issue
 When I attempt to stream a camera, I get no response. The logs indicate I am unable to get a default network address from homebridge/camera-utils (details below). I do get cached snapshots and that appears to work fine. 

### Screenshots/Logs
 [Ring] Preparing Live Stream for Side
[Ring] Failed to prepare stream for Side (0.408s)
[Ring] Error: Unable to get default network address
    at Object.<anonymous> (/Users/brandon/.hoobs/node_modules/@homebridge/camera-utils/lib/default-ip.js:25:19)
    at Generator.next (<anonymous>)
    at fulfilled (/Users/brandon/.hoobs/node_modules/@homebridge/camera-utils/lib/default-ip.js:5:58)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)

I get the same issue for all cameras (2 floodlight, 1 doorbell, 1 spotlight) using their appropriate names.

### Recommended solution
 If this is a homebridge issue and not ring issue I'm happy to submit it elsewhere, but it seems like @dgreif is the primary author regardless. I propose the cause of the issue is found and either fixed in code, or a root cause is described to fix on the user end. I'm happy to run debug commands/scripts as needed to help debug

### Environment

 - Ring Camera: 1x spotlightw_v2, 1x doorbell_v4, 2x floodlight_v2 
 - OS: OSX 10.15.4
 - Node.js: 12.18.3
 - NPM: 6.14.6
 - homebridge-ring (if applicable): 9.14.0
 - homebridge (if applicable): N/A (hoobs)
 - hoobs (if applicable): 3.3.1

### Network Environment
UniFi AP-AC-Pro
UniFi Switch 8 POE-150W
UniFi Security Gateway 3P

**Comment by adavison007** [comment]
@bc2297 Did it work prior to the update, or is this a fresh new install?

**Comment by bc2297** [comment]
@adavison007 fresh new install. First time using it

**Comment by adavison007** [comment]
> @adavison007 fresh new install. First time using it

What system are you using, HomebridgeUI, hoops etc? depending on what you have you may need to install FFmpeg , but i'm not and expert, soy may need @dgreif to help?

**Comment by dgreif** [comment]
He gave good details in the initial info.  I'm not sure why the code for default ip would fail on mac os, but it's actually not something that _must_ be fetched anymore.  I'll get out a fix soon that prevents the stream from failing if it can't get the default ip.  It's only there for backward compatibility at this point, and I'll remove it on my next major release (no plans to do one any time soon, but whenever I need to next)

**Comment by bc2297** [comment]
> What system are you using, HomebridgeUI, hoops etc?

Hoobs 3.3.1

> depending on what you have you may need to install FFmpeg , but i'm not and expert, soy may need @dgreif to help?

It was installed per the directions at https://github.com/homebridge/ffmpeg-for-homebridge

Looking at the source for the logging error (https://github.com/homebridge/camera-utils/blob/master/src/default-ip.ts) I don't see any ffmpeg calls and it seems like it might just be more networking related.

> He gave good details in the initial info. I'm not sure why the code for default ip would fail on mac os, but it's actually not something that must be fetched anymore. I'll get out a fix soon that prevents the stream from failing if it can't get the default ip. It's only there for backward compatibility at this point, and I'll remove it on my next major release (no plans to do one any time soon, but whenever I need to next)

Sounds great, thanks!

Do you have a one liner/example that I could use to print the values returned from the os/systeminfo calls in https://github.com/homebridge/camera-utils/blob/master/src/default-ip.ts ? Knowing what it's returning and looking at my networking settings might reveal what the issue is. Some of the possible things that come to mind are

- I have wifi turned off as ethernet is connected. Could it be trying to hit wifi first?
- I have a vpn profile installed on the computer which is generally unconnected. Could it be trying to get an IP from the VPN first?

As this is getting fixed anyway I realize it's not worth a ton of effort, so I'll put in some of my elbow grease if only for education's sake and patching my system

**Comment by bc2297** [comment]
@dgreif @adavison007 I was able to get things working correctly!
Background: I was running hoobs from the terminal as I had been previously debugging another plugin, so it wasn't running as a service.
Apparently my $PATH wasn't configured properly. I found this out when trying to find out the default network interface and not being able to run `route` in terminal. After updating my path, things work perfectly! Thanks for the help
